oak/server/rust/oak_loader/src/main.rs

```diff
@@ -70,24 +77,73 @@ pub struct Opt {
     introspect_port: u16,
     #[structopt(long, help = "Starts the Runtime without an introspection server.")]
     no_introspect: bool,
+    #[structopt(
+        long,
+        help = "Configuration files to expose to the Oak Application, each in key=filename format."
+    )]
+    config_files: Vec<ConfigEntry>,
+}
+
+/// A specification of a configuration entry as human readable key and a path to a file whose
+/// contents constitutes the actual value.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ConfigEntry {
+    key: String,
+    filename: String,
+}
+
+impl FromStr for ConfigEntry {
+    type Err = anyhow::Error;
+    fn from_str(v: &str) -> Result<Self, Self::Err> {
+        let parts = v.split("=").collect::<Vec<_>>();
+        if parts.len() != 2 {
+            return Err(anyhow!("could not parse config entry: {}", v));
+        }
+        Ok(ConfigEntry {
+            key: parts[0].to_string(),
+            filename: parts[1].to_string(),
+        })
+    }
 }
 
-fn read_file(filename: &str) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
+fn read_file(filename: &str) -> anyhow::Result<Vec<u8>> {
     let mut file = File::open(filename)
-        .map_err(|error| format!("Failed to open file <{}>: {:?}", filename, error))?;
+        .map_err(|error| anyhow!("Failed to open file <{}>: {:?}", filename, error))?;
     let mut buffer = Vec::new();
     file.read_to_end(&mut buffer)
-        .map_err(|error| format!("Failed to read file <{}>: {:?}", filename, error))?;
+        .map_err(|error| anyhow!("Failed to read file <{}>: {:?}", filename, error))?;
     Ok(buffer)
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
+fn parse_config_files(config_entries: &[ConfigEntry]) -> anyhow::Result<HashMap<String, Vec<u8>>> {
+    let mut file_map = HashMap::new();
+    for config_entry in config_entries {
+        let file_content = read_file(&config_entry.filename)?;
+        file_map.insert(config_entry.key.to_string(), file_content);
+    }
+    Ok(file_map)
+}
+
+pub fn parse_config_map(config_files: &[ConfigEntry]) -> anyhow::Result<ConfigMap> {
+    Ok(ConfigMap {
+        items: parse_config_files(config_files)?,
+    })
+}
+
+fn main() -> anyhow::Result<()> {
     if cfg!(feature = "oak_debug") {
         simple_logger::init_by_env();
     } else {
         eprintln!("No debugging output configured at build time");
     }
     let opt = Opt::from_args();
+    debug!("parsed opts: {:?}", opt);
+
+    let config_map = parse_config_map(&opt.config_files)?;
+    // We only log the keys here, since the values may be secret.
+    debug!("parsed config map entries: {:?}", config_map.items.keys());
+    // TODO(#689): Pass the `config_map` object to the Runtime instance, and make it available to
+    // the running Oak Application.
```

Comment:
I am not clear on how this will be used. Is the intention to make this available to wasm nodes? Or to dynamically replace tokens in the application configuration?

Reply:
The intention is to make it available to the first node of an Oak application, which can then take it apart and distribute it to further nodes if it needs to. It will be easier after #917 is merged, so I'll wait for that before doing the rest of the work.